docs/design-proposals/global-scheduler/GlobalSchedulerArchitectureDesignSpecificationFirstDraft.md

```diff
@@ -374,43 +373,114 @@ In our architecture, each scheduler has global view of resources and runs its sc
+The Scheduling Event Handler running inside this Home Scheduler process will do two things. It generates a key for this VM/Container request object and puts this object key into the work queue. It also saves the VM/Container request object into the Home Scheduler’s own Local Cache. The Scheduling Executor of this process will then dequeue the object key, retrieve the object from the cache, and run the scheduling algorithm to select the best cluster. After that, the Scheduling Executor will put the VM/Container’s scheduling decision into the Scheduler’s Scheduling Decision Queue and then goes back to process the next VM/Container request in the Work Queue. 
+Alternatively the Distributor can pass the VM/Container request to the selected Home Scheduler through Kubernetes object watching mechanism, i.e. the Distributor will save the object back into the VM/Container Info DB, which will trigger a notification to the API Server. The API server will then pass this object tagged with the selected Home Scheduler to the Home Scheduler process which has registered a watch through the “Reflector-Informer” mechanism. 
+No matter which mechanism is used, it introduces extra latency and implementation is much more complicated than the “thread” way.
+
+## 3.2 Scheduling Result Dispatching Flow
+![image.png](/images/3.2.png)
+
+This refers to the flow from Scheduling Decision Queue->VM/Container Request Dispatcher->Cluster. To ensure that the Scheduling Executor can process the VM/Container request in a timely manner, our design decouples the operation of the Dispatcher from the operation of the Scheduling Executor, i.e., the Dispatcher and the Scheduling Executor will run in separate threads. The Scheduling Executor puts the scheduling decision for a VM/Container into the Scheduling Decision Queue and immediately goes back to handle the next VM/Container scheduling. The Dispatcher will dequeue each VM/Container’s decision from this Scheduling Decision Queue and send the VM/Container request to the selected cluster. 
+All the security authentication/authorization as well as network error and retry during the communication with an external cluster, which may have varied latency, will be handled by the Dispatcher. If after a pre-defined number of retries, the Dispatcher still fails to send the VM/Container request to the selected cluster, the Dispatcher will put this VM/Container request back to the head of the Scheduling Work Queue and remove this cluster from the candidate list. 
+The Dispatcher will send the VM/Container request to the cluster via the VM/Container creation/deletion API over gRPC or HTTP2.   
+
+## 3.3 Scheduling Status Processing Flow
+This refers to the flow from Cluster->VM/Container Status Watcher->API Server->VM/Container Info DB. There are two ways to design and implement the scheduling status processing flow as described in the following two sections.
```

Comment:
I think  we don't  need to flow the status of VM/container.
The status change of  instances  can reflect by the resources status.

Reply:
As discussed in yesterday's meeting, the global scheduler does not handle scheduling success/failure. So this status processing flow will be removed.